spec/faraday/error_spec.rb

```diff
@@ -43,12 +43,26 @@
     end
 
     context 'maintains backward-compatibility until 1.0' do
-      it 'does not raise an error for nested error classes but prints an error message' do
+      it 'does not raise an error for error-namespaced classes but prints an error message' do
         error_message, error = with_warn_squelching { Faraday::Error::ClientError.new('foo') }
 
         expect(error).to be_a Faraday::ClientError
-        expect(error_message).to eq(
-          "DEPRECATION WARNING: Faraday::Error::ClientError is deprecated! Use Faraday::ClientError instead.\n"
+        expect(error_message).to match(
+          Regexp.new(
+            'NOTE: Faraday::Error::ClientError.new is deprecated; '\
+            'use Faraday::ClientError.new instead. It will be removed in or after version 1.0'
+          )
+        )
+      end
+
+      it 'does not raise an error for inherited error-namespaced classes but prints an error message' do
+        error_message, = with_warn_squelching { class E < Faraday::Error::ClientError; end }
```

Comment:
this looks like a typo, but it's not. Rubocop likes `a, = [:foo, :bar]` better than `a, _ = [:foo, :bar]` 🤷‍♂